components/brave_news/browser/peeking_card_unittest.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_news/browser/peeking_card.h"
+
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "brave/components/brave_news/browser/feed_sampling.h"
+#include "brave/components/brave_news/common/brave_news.mojom.h"
+#include "brave/components/brave_news/common/subscriptions_snapshot.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_news {
+
+namespace {
+
+ArticleInfo FakeInfo(std::string id = "",
+                     std::vector<std::string> channels = {},
+                     GURL url = GURL()) {
+  static int next_id = 0;
+
+  if (id.empty()) {
+    id = base::NumberToString(++next_id);
+  }
+
+  auto item = mojom::FeedItemMetadata::New();
+  item->publisher_id = id;
+  item->channels = channels;
+  item->url = url;
+
+  return std::make_tuple(std::move(item), ArticleMetadata());
+}
+
+}  // namespace
+
+TEST(BraveNewsPeekingCard, NoArticles) {
+  EXPECT_EQ(std::nullopt, PickPeekingCard(SubscriptionsSnapshot(), {}, {}));
+}
+
+TEST(BraveNewsPeekingCard, NoSubscribedArticles) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+  EXPECT_EQ(std::nullopt,
+            PickPeekingCard(SubscriptionsSnapshot(), {}, articles));
+}
+
+TEST(BraveNewsPeekingCard, PublishersAreIncluded) {
+  ArticleInfos articles;
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+  articles.push_back(FakeInfo());
+
+  size_t enabled_index = 1;
+  SubscriptionsSnapshot subscriptions(
+      {std::get<0>(articles[enabled_index])->publisher_id}, {}, {}, {});
+
+  EXPECT_EQ(enabled_index,
+            PickPeekingCardWithMax(subscriptions, {}, articles, 1));
```

Comment:
Curious why `1` is passed explicitely as max candidate value in this file?

Reply:
Ah, because we pick a random candidate from the top `N` - using 1 makes it much easier to predict the results

Reply:
Ah, Ok. We want to ignore the randomness from this test.